var/spack/repos/builtin/packages/chapel/package.py

```diff
@@ -3,24 +3,385 @@
+
+    depends_on("doxygen@1.8.17:")
+
+    variant(
+        "llvm",
+        default="unset",
+        description="LLVM backend type. Use value 'spack' to have spack "
+        "handle the LLVM package",
+        values=("none", "system", "bundled", "spack", "unset"),
+    )
+
+    variant(
+        "comm",
+        default="none",
+        description="Build Chapel with multi-locale support",
+        values=("none", "gasnet", "ofi"),
+    )
+    variant(
+        "substrate",
+        default="none",
+        description="Build Chapel with mulit-locale support using the "
+        "supplied CHPL_COMM_SUBSTRATE",
+        values=("none", "udp", "ibv", "ofi"),
+        multi=False,
+    )
+
+    package_module_opts = ("zmq", "libevent", "protobuf", "ssl", "hdf5", "yaml")
+    package_module_dict = {
+        "zmq": "libzmq",
+        "libevent": "libevent",
+        "protobuf": "protobuf",
+        "ssl": "openssl",
+        "hdf5": "hdf5+hl~mpi",
+        "yaml": "libyaml",
+    }
+
+    variant(
+        "package_modules",
+        description="Include package module dependencies with spack",
+        values=disjoint_sets(("none",), ("all",), package_module_opts)
+        .with_error("'none' or 'all' cannot be activated along with other package_modules")
+        .with_default("none")
+        .with_non_feature_values("none", "all"),
+    )
+
+    for opt, dep in package_module_dict.items():
+        depends_on(dep, when="package_modules={0}".format(opt), type="run")
+        depends_on(dep, when="package_modules=all", type="run")
+
+    platform_opts = (
+        "unset",
+        "cygwin32",
+        "cygwin64",
+        "darwin",
+        "linux32",
+        "linux64",
+        "netbsd32",
+        "netbsd64",
+        "pwr6",
+        "cray-cs",
+        "cray-xc",
+        "hpe-apollo",
+        "hpe-cray-ex",
+    )
+
+    variant(
+        "host_platform",
+        description="Host platform",
+        default="unset",
+        values=platform_opts,
+        multi=False,
+    )
+
+    variant(
+        "target_platform",
+        description="Target platform for cross compilation",
+        default="unset",
+        values=platform_opts,
+        multi=False,
+    )
+
+    variant(
+        "tasks",
+        description="Select tasking layer for intra-locale parallelism",
+        default="qthreads",
+        values=("qthreads", "fifo"),
+        multi=False,
+    )
+
+    variant(
+        "re2",
+        description="Build with re2 support",
+        default="bundled",
+        values=("none", "bundled"),
+        multi=False,
+    )
+
+    variant(
+        "gmp",
+        description="Build with gmp support",
+        default="unset",
+        values=("unset", "system", "none", "bundled"),
+        multi=False,
+    )
+
+    variant(
+        "hwloc",
+        description="Build with hwloc support",
+        default="bundled",
+        values=("none", "bundled"),
+        multi=False,
+    )
+
+    variant(
+        "aux_filesys",
```

Comment:
Pretty sure this is deprecated and doesn't need to be in the Spack package here

Reply:
Oh, I didn't see that it was marked deprecated, our environment setup docs might need an update in that case.